More resilient conversions from strings

pashadia opened the pull request:
* Added case-insensitive conversions, so that e.g. `Card::from("AH")` works
* Added clearer error messages when the conversion fails

pashadia committed:
Implement case-insensitive conversion

pashadia committed:
Add clearer conversion error messages

Nydauron reviewed (commented):
Thanks for the PR! This looks good for the most part.

Could you add some documentation to `Suit::from_char` explaining that the character is case-insensitive?

Also, CI failed to run due to clippy failing (assumably because some clippy rule was updated). I already went ahead and reordered the CI steps and fixed the clippy error on master. As I was writing this, I saw you made a push fixing it on this feature branch, and while I appreciate the change, I'd suggest popping off that commit and making that a separate PR.

pashadia committed:
Add doc string for Suit::from_char()

pashadia committed:
Merge branch 'Nydauron:master' into master

pashadia commented:
Hey!

I've removed the CI-related commit, and I've added the documentation to the `Suit::from_char()` function.

As I've done that, I see that I've done the case-insensitive implementation for the Value conversion in the `impl TryFrom<char> for Value` block, instead of inside `Value::from_char()`. Functionally, it's the same thing, but I think these should have similar implementations --  would you like me to change the `Value` impl so that it's similar to the `Suit` ?

pashadia committed:
Implement Value::from_char similarly to Suit

pashadia commented:
Rewritten the `Value` conversion from char in a similar fashion to `Suit`. Let me know if you have any comments.

Nydauron reviewed (approved):
LGTM, thanks!
